src/middlewares/promises.js

```diff
@@ -0,0 +1,18 @@
+const middleware = store => next => action => {
+	if (action.type !== 'PROMISE') {
+		return next(action);
+	}
+	const [startAction, successAction, failureAction] = action.actions;
+	store.dispatch({
+		type: startAction,
+	});
+	action.payload.then((data) => store.dispatch({
+		type: successAction,
+		data,
+	}), (error) => store.dispatch({
+		type: failureAction,
+		error,
+	}));
+};
+
+export default middleware;
```

Comment:
what happened if result of function from 9 line will be faster than from 6?
You should chain all 3 actions, not only success => error